integrations/google-analytics-4/lib/index.js

```diff
@@ -0,0 +1,310 @@
+'use strict';
+
+/**
+ * Module dependencies.
+ */
+var integration = require('@segment/analytics.js-integration');
+var reject = require('reject');
+
+/**
+ * GA4
+ */
+var GA4 = (module.exports = integration('Google Analytics 4')
+  .global('gtag')
+  .global('ga4DataLayer')
+  .option('measurementIds', [])
+  .option('cookieDomainName', '')
+  .option('cookiePrefix', '')
+  .option('cookieExpiration', 0)
+  .option('cookieUpdate', true)
+  .option('cookieFlags', '')
+  .option('disablePageViewMeasurement', true)
+  .option('disableAllAdvertisingFeatures', false)
+  .option('disableAdvertisingPersonalization', false)
+  .option('disableGoogleAnalytics', false)
+  .option('sendUserId', false)
+  .option('userProperties', {})
+  .option('customEventsAndParameters', [])
+  .tag(
+    '<script src="//www.googletagmanager.com/gtag/js?id={{ measurementId }}&l=ga4DataLayer">'
+  ));
+
+/**
+ * Initialize.
+ *
+ * https://developers.google.com/analytics/devguides/collection/ga4
+ *
+ * @api public
+ */
+GA4.prototype.initialize = function() {
+  window.ga4DataLayer = window.ga4DataLayer || [];
+  window.gtag = function() {
+    window.ga4DataLayer.push(arguments);
+  };
+
+  // This line is in all of the gtag examples but is not well documented. Research says
+  // says that it is is related to deduplication.
+  // https://stackoverflow.com/questions/59256532/what-is-the-js-gtags-js-command
+  window.gtag('js', new Date());
+
+  var config = [];
+  var opts = this.options;
+  var measurementIds = opts.measurementIds;
+
+  // Avoid loading and configuring gtag.js if any are true:
+  //   - Disable Google Analytics setting is enabled
+  //   - No measurement IDs are configured
+  if (!measurementIds.length || opts.disableGoogleAnalytics) {
+    return;
+  }
+
+  // Measurement IDs
+  // https://developers.google.com/analytics/devguides/collection/ga4#add_an_additional_google_analytics_property_to_an_existing_tag
+  for (var i = 0; i < measurementIds.length; i++) {
+    config.push(['config', measurementIds[i]]);
+  }
+
+  // Disable Page View Measurement
+  // https://developers.google.com/analytics/devguides/collection/ga4/disable-page-view
+  if (opts.disablePageViewMeasurement) {
+    for (var i = 0; i < measurementIds.length; i++) {
+      config.push([
+        'config',
+        measurementIds[i],
+        {
+          send_page_view: false
+        }
+      ]);
+    }
+  }
```

Comment:
It looks like we add the GA properties then if they have another setting enabled ie `disablePageViewMeasurement` push the same properties to the config with with `send_page_view: false` Curious if this causes any dupes? Looks like config is just an array

Reply:
Not sure if this would cause dupes. I think it would be cleaner to push the properties once, so I can make that change.

Reply:
Thank you so much for calling this out! Seems when we call `config` multiple time may overwrite others 🤔.